### PR TITLE
Index allo v2 rounds

### PR DESCRIPTION
**Pull request by gravityblast**
This PR adds indexing of Allo V2 pools converted to rounds.
First part of https://github.com/gitcoinco/grants-stack-indexer/issues/383

The next PR will insert and manager pending roles and roles for rounds.

### CODE REVIEW

File: src/indexer/allo/v2/handleEvent.ts [hunk 1]
```diff
@@ -1,10 +1,36 @@
...
+function padBytes32Hex(s: string): string {
+  if (s.length > 64) {
+    return s;
+  }
+
+  const padding = 64 - s.length;
+  let hex = s;
+  hex = "0".repeat(padding) + hex;
+  return "0x" + hex;
+}
```
Comment (vacekj) [inline src/indexer/allo/v2/handleEvent.ts:20]: We can use https://viem.sh/docs/utilities/pad#pad here

File: src/indexer/allo/v2/handleEvent.ts [hunk 2]
```diff
@@ -1,10 +1,36 @@
...
+
+function generateRoundRoles(poolId: string) {
+  // POOL_MANAGER_ROLE = bytes32(poolId);
+  const managerRole = padBytes32Hex(poolId);
+
+  // POOL_ADMIN_ROLE = keccak256(abi.encodePacked(poolId, "admin"));
+  const adminRawRole = ethers.utils.solidityPack(
+    ["uint256", "string"],
+    [poolId, "admin"]
```
Comment (vacekj) [inline src/indexer/allo/v2/handleEvent.ts:29]: https://viem.sh/docs/abi/encodePacked#encodepacked

File: src/indexer/allo/v2/handleEvent.ts [hunk 3]
```diff
@@ -1,10 +1,36 @@
...
+  );
+  const adminRole = ethers.utils.solidityKeccak256(["bytes"], [adminRawRole]);
```
Comment (vacekj) [inline src/indexer/allo/v2/handleEvent.ts:31]: https://viem.sh/docs/utilities/keccak256#keccak256

**Review by vacekj — APPROVED** [review]
couple changes requested otherwise LGTM.

File: src/indexer/allo/v2/parsePoolMetadata.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
+type ipfsGetFn = <T>(cid: string) => Promise<T | undefined>;
```
Comment (boudra) [inline src/indexer/allo/v2/parsePoolMetadata.ts:1]:
```suggestion
type IpfsGetFn = <T>(cid: string) => Promise<T | undefined>;
```

types are title case

Comment (boudra) [reply src/indexer/allo/v2/parsePoolMetadata.ts:1]: not sure if the name of the file makes sense with the contents, maybe just `poolMetadata.ts`?

**Review by boudra — APPROVED** [review]

File: src/indexer/allo/v2/handleEvent.ts (hunk 1 above)
Comment (gravityblast) [reply src/indexer/allo/v2/handleEvent.ts:20]: @vacekj good catch, I'm going to change it in branch `allo-v2-rounds-4` where I fixed that part converting the poolId to an hex string

File: src/indexer/allo/v2/parsePoolMetadata.ts (hunk 4 above)
Comment (gravityblast) [reply src/indexer/allo/v2/parsePoolMetadata.ts:1]: @boudra  update

File: src/indexer/allo/v2/handleEvent.ts (hunk 3 above)
Comment (gravityblast) [reply src/indexer/allo/v2/handleEvent.ts:31]: cool, I updated it thank you!

File: src/indexer/allo/v2/handleEvent.ts (hunk 2 above)
Comment (gravityblast) [reply src/indexer/allo/v2/handleEvent.ts:29]: done as well thank you!

**Comment by gravityblast** [comment]
@vacekj @boudra I pushed the changes to `allo-v2-rounds-4`

**Pull request merged**